lib/inc/internal/tokenizer.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "tokens.hpp"
+
+#include <boost/nowide/fstream.hpp>
+#include <vector>
+#include <queue>
+#include <string>
+
+namespace hocon {
+
+    // This exception should not leave this file
+    class problem_exception : std::runtime_error {
+    public:
+        problem_exception(problem prob);
+        problem const& get_problem() const;
+
+    private:
+        problem _problem;
+    };
+
+    class config_exception : public std::runtime_error {
```

Comment:
No it's fine to move it, I just put it in the only place it'd come up so far. The exception handling in this class is also most likely not in its final form yet, because I didn't understand what they were doing with it. The tests gave some hint, but I want to see it in action elsewhere before I switch everything over to using problem_exception, since I think the implementation will be a bit different than the Java.